### PR TITLE
Docs: Update documentation after introduction of noexcept

### PR DESCRIPTION
**Pull request by matusvalo**
- [x] Update language basics pure python
- [x] Update tutorial

### CODE REVIEW

**Comment by matusvalo** [comment]
Tutorial finished. Marking as ready for review.

**Comment by matusvalo** [comment]
Part of this PR is pushed to #5094 - https://github.com/cython/cython/pull/5094/files#diff-186acf91c9efce0ce6d0762e02778681791b6d88b11d56ecdd876cf178e0604d . Not sure if it leave it there or it should be moved to this PR...

**Comment by da-woods** [comment]
> Not sure if it leave it there or it should be moved to this PR...

Probably OK to leave it there.

File: docs/src/tutorial/clibraries.rst [hunk 1]
```diff
@@ -608,39 +607,27 @@ Note how we have effectively created a fast path through the method in
...
-exception.
+calling code.
 
 The only way calling code can deal with this situation is to call
 ``PyErr_Occurred()`` when returning from a function to check if an
 exception was raised, and if so, propagate the exception.  This
-obviously has a performance penalty.  Cython therefore allows you to
-declare which value it should implicitly return in the case of an
+obviously has a performance penalty.  Cython therefore uses a dedicated value
+it should implicitly return in the case of an
```
Comment (da-woods) [inline docs/src/tutorial/clibraries.rst:622]:
```suggestion
obviously has a performance penalty.  Cython therefore uses a dedicated value
that it implicitly returns in the case of an
```

**Review by da-woods — COMMENTED** [review]
Looks good - I suggested one small language change.

**Comment by matusvalo** [comment]
Comment fixed.

**Comment by da-woods** [comment]
thanks

**Pull request merged**